Source/Pharmacist/HarmonyPatches/WorkGiver_DoBill_AddEveryMedicineToRelevantThings.cs

```diff
@@ -0,0 +1,41 @@
+﻿// WorkGiver_DoBill_GetMedicalCareCategory.cs
+// Copyright Karel Kroeze, 2018-2018
+
+using HarmonyLib;
+using RimWorld;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Verse;
+using Verse.AI;
+
+namespace Pharmacist.Properties
+{
+    [HarmonyPatch(typeof(WorkGiver_DoBill), "AddEveryMedicineToRelevantThings")]
+    public static class WorkGiver_DoBill_AddEveryMedicineToRelevantThings
+    {
+        public static bool Prefix(Pawn pawn, Thing billGiver, List<Thing> relevantThings, Predicate<Thing> baseValidator, Map map)
+        {
+            Pawn patient = billGiver as Pawn;
+            //MedicalCareCategory medicalCareCategory = GetMedicalCareCategory(billGiver);
+            MedicalCareCategory medicalCareCategory = PharmacistUtility.TendAdvice(patient, InjurySeverity.Operation);
+            List<Thing> list = map.listerThings.ThingsInGroup(ThingRequestGroup.Medicine);
+            List<Thing> tmpMedicine = (List<Thing>)(typeof(WorkGiver_DoBill).GetField("tmpMedicine", BindingFlags.NonPublic | BindingFlags.Static).GetValue(null));
+            tmpMedicine.Clear();
+            for (int i = 0; i < list.Count; i++)
+            {
+                Thing thing = list[i];
+                if (medicalCareCategory.AllowsMedicine(thing.def) && baseValidator(thing) && pawn.CanReach(thing, PathEndMode.OnCell, Danger.Deadly))
+                {
+                    tmpMedicine.Add(thing);
+                }
+            }
+            float inverter = 1f;
+            if (patient.BillStack.FirstShouldDoNow.recipe.defName == "Anesthetize") { inverter = -1f; }
+            tmpMedicine.SortBy((Thing x) => 0f - (x.GetStatValue(StatDefOf.MedicalPotency) * inverter), (Thing x) => x.Position.DistanceToSquared(billGiver.Position));
```

Comment:
The `inverter` variable in the above 3 lines is the only place we intend to affect any change here.

Reply:
heya, I know it's been ages since you made this PR, but did you check if this affects op chances? Don't more advanced meds increase success chances? Because pawns will pick up stacks of meds at once, if this affects operations in that they pick up crappy meds for anestesis _and_ whatever affects the op itself, that might be an issue.